### PR TITLE
Add EIP-5437: Interface for Security Contact

### PR DESCRIPTION
**Pull request by xinbenlv**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-sc.md
Comment (Pandapip1) [inline EIPS/eip-sc.md:2]:
```suggestion
eip: 5437
```

File: EIPS/eip-sc.md
Comment (SamWilsn) [inline EIPS/eip-sc.md:20]:
```suggestion
## Abstract

An interface for smart contracts to specify how to communicate with their developers using asymmetric encryption. For example, a security researcher may wish to report vulnerabilities over a secured channel. The interface exposes an asymmetric encryption key and a destination for delivery.
```

File: EIPS/eip-sc.md
Comment (SamWilsn) [inline EIPS/eip-sc.md:30]: Generally interfaces are named using `PascalCase` (so no underscores.)

File: EIPS/eip-sc.md
Comment (SamWilsn) [inline EIPS/eip-sc.md:40]: I'd recommend either splitting these optional functions into their own interface(s), or defining what they return when not implemented.

File: EIPS/eip-sc.md
Comment (SamWilsn) [inline EIPS/eip-sc.md:33]: Why does this take an argument?

I might also suggest the name `securityContact` instead.

File: EIPS/eip-sc.md
Comment (SamWilsn) [inline EIPS/eip-sc.md:44]: I think you can omit setting the security contact from the interface entirely. It'll be project dependent anyway.

File: EIPS/eip-sc.md
Comment (SamWilsn) [inline EIPS/eip-sc.md:45]: I would suggest the name `SecurityContactChanged` instead.

File: EIPS/eip-sc.md
Comment (SamWilsn) [inline EIPS/eip-sc.md:53]: I think this can be left out entirely, and possibly done as a separate EIP.

File: EIPS/eip-sc.md
Comment (SamWilsn) [inline EIPS/eip-sc.md:66]: Would it also make sense to support a secp256k1 gpg algorithm?

File: EIPS/eip-sc.md
Comment (xinbenlv) [reply EIPS/eip-sc.md]: Done! thank you!

File: EIPS/eip-sc.md
Comment (xinbenlv) [reply EIPS/eip-sc.md]: Thank you, update according to your revision.

File: EIPS/eip-sc.md
Comment (xinbenlv) [reply EIPS/eip-sc.md]: Updated to IEIP5437

File: EIPS/eip-sc.md
Comment (xinbenlv) [reply EIPS/eip-sc.md]: Sounds good, would you be ok if I leave it to a follow up revision but to get this file check in first

File: EIPS/eip-sc.md
Comment (xinbenlv) [reply EIPS/eip-sc.md]: I will provide detail in rationale for why there is an argument

File: EIPS/eip-sc.md
Comment (xinbenlv) [reply EIPS/eip-sc.md]: That seems reasonable, I'd leave it here to triger design discussion for now. May remove when we decide not to have it. One argument would be regular rotation of public keys

File: EIPS/eip-sc.md
Comment (xinbenlv) [reply EIPS/eip-sc.md]: Done, thanks for the suggestion

File: EIPS/eip-sc.md
Comment (xinbenlv) [reply EIPS/eip-sc.md]: That sounds a good idea. Let me think about it. Added a TODo

File: EIPS/eip-sc.md
Comment (xinbenlv) [reply EIPS/eip-sc.md]: I was actually thinking of it! Because RSA keys were so long and inefficient. Added a TODO

File: EIPS/eip-5437.md
Comment (Pandapip1) [inline EIPS/eip-5437.md:81]: This link will not be allowed

Comment (xinbenlv) [reply EIPS/eip-5437.md]: Updated to "add proper link to openstandard"

File: EIPS/eip-5437.md
Comment (Pandapip1) [inline EIPS/eip-5437.md:80]:
```suggestion

```

No external links.

File: EIPS/eip-5437.md
Comment (Pandapip1) [inline EIPS/eip-5437.md:162]:
```suggestion
Currently, existing solutions such as OpenZeppelin use plaintext in source code
```
PlainText™ is a registered trademark of the NonExistent corporation. Used without permission for laughs.

(/j obviously)

File: EIPS/eip-5437.md
Comment (Pandapip1) [inline EIPS/eip-5437.md:169]:
```suggestion
It's recommend that new versions of smart contracts adopt this EIP in addition to the legacy `@custom:security-contact` approach.
```

File: EIPS/eip-5437.md
Comment (Pandapip1) [inline EIPS/eip-5437.md:175]:
```suggestion
Implementors should properly follow security practices required by the encryption scheme to ensure the security of the chosen communication channel. Some best practices are as follows:
```

File: EIPS/eip-5437.md
Comment (Pandapip1) [inline EIPS/eip-5437.md:180]:
```suggestion
3. Regularly monitor the channel to receive notices in a timely manner.
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
Please also run this through a spell/grammar checker. I've almost certainly missed a few typos or other errors.

**Comment by xinbenlv** [comment]
All suggestions committed. This commit mean to check in the file as Draft. I will run through grammar and spelling check prior to moving to Review status

**Comment by xinbenlv** [comment]
Request a manual merge @SamWilsn

**Comment by Pandapip1** [comment]
Why does this need a manual merge?

File: EIPS/eip-5437.md
Comment (Pandapip1) [reply EIPS/eip-5437.md:65]: My previous comment needs fixing.

File: EIPS/eip-5437.md
Comment (Pandapip1) [reply EIPS/eip-5437.md:72]: My previous comment needs fixing here too.

File: EIPS/eip-5437.md
Comment (Pandapip1) [reply EIPS/eip-5437.md:120]: My previous comment needs fixing here too.

File: EIPS/eip-5437.md
Comment (Pandapip1) [reply EIPS/eip-5437.md:126]: My previous comment needs fixing here too.

File: EIPS/eip-5437.md
Comment (Pandapip1) [reply EIPS/eip-5437.md:132]: My previous comment needs fixing here too.

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
Previous comments need fixing

File: EIPS/eip-5437.md
Comment (xinbenlv) [reply EIPS/eip-5437.md:62]: Technical answer: While I appreciate your suggestion: yes we are still considering using EC-based algorithm but not determine yet.

I feel this is not editorial comment and shall not block merge of the file.

File: EIPS/eip-5437.md
Comment (xinbenlv) [reply EIPS/eip-5437.md:72]: I incline to keep it as is: 

secp256k1 is being considered but not rule out RSA/3072 for it's the default of OpenGPG. I don't think an technical but non-editorial comment shall block merging a DRAFT

File: EIPS/eip-5437.md
Comment (xinbenlv) [reply EIPS/eip-5437.md:65]: Let me reword

Comment (xinbenlv) [reply EIPS/eip-5437.md]: DONE

File: EIPS/eip-5437.md
Comment (xinbenlv) [reply EIPS/eip-5437.md:122]: I hold a different opinion in this. This is the last way of emergency notification or notification to DAO that doesn't have a centralized authority but still need to be notified.

File: EIPS/eip-5437.md
Comment (xinbenlv) [reply EIPS/eip-5437.md:130]: Thank you for this suggestion but I don't prefer this markdown style change. We don't have a particular flavor of markdown style.

File: EIPS/eip-5437.md
Comment (xinbenlv) [reply EIPS/eip-5437.md:126]: Let's leave it a TODO

File: EIPS/eip-5437.md
Comment (Pandapip1) [inline EIPS/eip-5437.md:122]:
```suggestion
4. It's also RECOMMENDED that an on-chain security notify method `securityNotify` to implemented to receive security notice onchain. If it's implemented and a call has succeeded, it MUST emit an `OnSecurityNotification` with identical pass-in-parameter data.
```

File: EIPS/eip-5437.md
Comment (Pandapip1) [inline EIPS/eip-5437.md:130]:
```suggestion
6. It's recommended to set a bounty policy via `bountyPolicy` method. The `id = 0` is preserved for a full overview, while other digits are used for different individual bounty policies. The returned string will be URI to content of bounty policies. No particular format of bounty policy is specified.
```

File: EIPS/eip-5437.md
Comment (Pandapip1) [inline EIPS/eip-5437.md:68]:
```suggestion
2. The `publicKey` returned from `getSecurityContact` MUST follow the encryption scheme specified in the table above.
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
There are still these formatting errors that need fixing.

These result in inconsistent rendering and are therefore *actually* blockers, not stylistic choices.

**Review by Pandapip1 — APPROVED** [review]
As much as I feel that there are issues with this, it's valid for a draft.

**Pull request merged**
